### PR TITLE
chore: github action to deploy on prod

### PR DESCRIPTION
**Pull request by MKCG**
This is the final changeset of the migration for the API. The commits will be squashed when merging into the main branch.

The PR add an automated deployment of the API on production.

It also : 
- disable the query explaination when not in dev env for perf reason
- remove two args (`skip_empty` & `include_meta`) that were generating HTTP 500 even though the installed pip modules are identical

### CODE REVIEW

File: aio/aio-proxy/aio_proxy/search/helpers/helpers.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 
 def extract_ul_and_etab_from_es_response(structure):
-    structure_dict = structure.to_dict(skip_empty=False, include_meta=False)
```
Comment (HAEKADI) [inline aio/aio-proxy/aio_proxy/search/helpers/helpers.py:5]: Why did you remove `skip_empty=False` ?

File: aio/aio-proxy/aio_proxy/search/helpers/color.py [hunk 2]
```diff
@@ -2,12 +2,15 @@
...
 
-from aio_proxy.response.helpers import COLOR_URL
+from aio_proxy.response.helpers import COLOR_URL, is_dev_env
 
 logging.basicConfig(level=logging.INFO)
 
 
 def get_current_color(color_url):
+    if is_dev_env() is False:
+        return "reader"
```
Comment (HAEKADI) [inline aio/aio-proxy/aio_proxy/search/helpers/color.py:12]: I don't quite understand the "reader" logic. What is the point of this string ?

File: aio/aio-proxy/aio_proxy/search/es_search_runner.py [hunk 3]
```diff
@@ -69,9 +70,14 @@ def execute_and_format_es_search(self):
             self.es_search_results.append(matching_structure_dict)
 
     def sort_and_execute_es_search_query(self):
-        self.es_search_client = self.es_search_client.extra(
-            track_scores=True, explain=True
-        )
+        self.es_search_client = self.es_search_client.extra(track_scores=True)
+
+        # explain query result in dev env
```
Comment (HAEKADI) [inline aio/aio-proxy/aio_proxy/search/es_search_runner.py:75]: I actually use the `score` in all environments.

**Review by HAEKADI — CHANGES_REQUESTED** [review]

File: aio/aio-proxy/aio_proxy/search/helpers/color.py (hunk 2 above)
Comment (MKCG) [reply aio/aio-proxy/aio_proxy/search/helpers/color.py:12]: The github workflow is not compatible with the "siren-reader" alias since the dev server has not been migrated yet.

Comment (HAEKADI) [reply aio/aio-proxy/aio_proxy/search/helpers/color.py:12]: I see.

File: aio/aio-proxy/aio_proxy/search/helpers/helpers.py (hunk 1 above)
Comment (MKCG) [reply aio/aio-proxy/aio_proxy/search/helpers/helpers.py:5]: As explained in the description of the pull request, I had to remove those args to make it work.
Otherwise the app was raising this error : https://errors.data.gouv.fr/organizations/sentry/issues/132881/?project=14&query=is%3Aunresolved+is%3Afor_review+assigned_or_suggested%3A%5Bme%2C+none%5D&referrer=issue-stream&sort=inbox

**Review by XavierJp — APPROVED** [review]

**Comment by HAEKADI** [comment]
@MKCG I see that the `deploy` workflow runs independently if the test workflow. I think we should only deploy once that tests have been successful, e2e tests that is. What do you think ?

**Comment by MKCG** [comment]
> @MKCG I see that the `deploy` workflow runs independently of the test workflow. I think we should only deploy once that tests have been successful, e2e tests that is. What do you think ?

@HAEKADI , I agree. The test pipeline is executed on the dev server, right ? We first need to make sure that the dev server API is up to date.

**Comment by HAEKADI** [comment]
Yes, the test workflow executed on GitHub Actions is based on the indices in `dev`, so I have to always make sure that `siren-blue` is up to date since that is the index I use by default here.
I suppose we'll have to change the way we run tests on Github Actions and use `index-alias` instead.

**Comment by HAEKADI** [comment]
> > @MKCG I see that the `deploy` workflow runs independently of the test workflow. I think we should only deploy once that tests have been successful, e2e tests that is. What do you think ?
> 
> @HAEKADI , I agree. The test pipeline is executed on the dev server, right ? We first need to make sure that the dev server API is up to date.

Yes, the test workflow executed on GitHub Actions is based on the indices in `dev`, so I have to always make sure that `siren-blue` is up to date since that is the index I use by default here.
I suppose we'll have to change the way we run tests on Github Actions and use `index-alias` instead.

**Review by HAEKADI — APPROVED** [review]
🚀
